lib/puppet/provider/mongodb.rb

```diff
@@ -40,17 +40,39 @@ def self.mongo_eval(cmd, db = 'admin')
     # config files.
     config = YAML.load_file(file)
     if config.kind_of?(Hash) # Using a valid YAML file for mongo 2.6
+      bindip = config['net.bindIp']
       port = config['net.port']
+      shardsvr = config['sharding.clusterRole']
+      confsvr = config['sharding.clusterRole']
     else # It has to be a key-value config file
       config = {}
       File.readlines(file).collect do |line|
          k,v = line.split('=')
          config[k.rstrip] = v.lstrip.chomp if k and v
       end
+      bindip = config['bind_ip']
       port = config['port']
+      shardsvr = config['shardsvr']
+      confsvr = config['confsvr']
     end
 
-    out = mongo([db, '--quiet', '--port', port, '--eval', cmd])
+    if bindip and ! bindip.eql? '0.0.0.0'
```

Comment:
This seems to replicate the functionality in mongodb_replset/mongo.rb, could that be condensed?